src/docfx/cli/CommandLine.cs

```diff
@@ -0,0 +1,125 @@
+            {
+                Handler = CommandHandler.Create((CommandLineOptions options) => Parse(name, options)),
+            };
+
+        private Command NewCommand()
+        {
+            var command = BaseCommand("new");
+
+            command.AddOption(new Option<string>(
+                new[] { "-o", "--output" }, "Output directory in which to place built artifacts."));
+            command.AddOption(new Option<bool>(
+                "--force", "Forces content to be generated even if it would change existing files."));
+            command.AddArgument(new Argument<string>("templateName", "Docset template name"));
+            return command;
+        }
+
+        private Command RestoreCommand()
+        {
+            var command = BaseCommand("restore");
+            DefineCommonCommands(command);
+            return command;
+        }
+
+        private Command BuildCommand()
+        {
+            var command = BaseCommand("build");
+            DefineCommonCommands(command);
+
+            command.AddOption(new Option<string[]>(
+                new[] { "--file", "--files" }, "Build only the specified files."));
+            command.AddOption(new Option<string>(
+                new[] { "-o", "--output" }, "Output directory in which to place built artifacts."));
+            command.AddOption(new Option<OutputType>(
+                "--output-type", "Output directory in which to place built artifacts."));
+            command.AddOption(new Option<bool>(
+                "--dry-run", "Do not produce build artifact and only produce validation result."));
+            command.AddOption(new Option<bool>(
+                "--no-dry-sync", "Do not run dry sync for learn validation."));
+            command.AddOption(new Option<bool>(
+                "--no-restore", "Do not restore dependencies before build."));
+            command.AddOption(new Option<bool>(
+                "--no-cache", "Always fetch latest dependencies in build."));
+            command.AddOption(new Option<string>(
+                "--template-base-path", "The base path used for referencing the template resource file when applying liquid."));
+            return command;
+        }
+
+        private Command ServeCommand()
+        {
+            var command = BaseCommand("serve");
+            DefineCommonCommands(command);
+
+            command.AddOption(new Option<bool>(
+                "--language-server", "Do not produce build artifact and only produce validation result."));
+            command.AddOption(new Option<bool>(
+                "--no-cache", "Always fetch latest dependencies in build."));
+            return command;
+        }
+
+        private void Parse(string command, CommandLineOptions options) => (_command, _options) = (command, options);
+
+        private static void DefineCommonCommands(Command command)
+        {
+            command.AddArgument(new Argument<string>("WorkingDirectory", () => "."));
```

Comment:
Previously this is called `directory` with a description.

```
syntax.DefineParameter("directory", ref options.WorkingDirectory, "A directory that contains docfx.yml/docfx.json.");
```

Reply:
And https://github.com/dotnet/command-line-api/issues/1156 shows to use `{ Arity = ArgumentArity.ZeroOrOne }` to hide the wired `[default:]` thing.

Reply:
Same is true for `new` command, previously `docfx new` shows a template list, now it causes an error because `templateName` is now a required argument.